System/Random.hs

```diff
@@ -1083,6 +1084,29 @@ uniformIntegerM (l, h) gen
         let v' = v * b + fromIntegral x
         v' `seq` f (mag * b) v'
 
+-- | A slightly modified version of what can be found in [Lemire's
+-- paper](https://arxiv.org/pdf/1805.10941.pdf), [O'Neill's
+-- blogpost](https://www.pcg-random.org/posts/bounded-rands.html) and
+-- more directly from [O'Neill's github
+-- repo](https://github.com/imneme/bounded-rands/blob/3d71f53c975b1e5b29f2f3b05a74e26dab9c3d84/bounded32.cpp#L234).
+-- The modification is we want the range to be [0,t] not [0,t) to be
+-- compatible with unsignedBitmaskWithRejectionRM.
```

Comment:
I hoisted the `+1` up into the caller. That makes the algorithm itself more similar to what's described in the literature, and saved me from having to write a lengthy comment about how the algorithm is different.

https://github.com/idontgetoutmuch/random/pull/66/files#diff-4a522d098452ebfc9be2d6291588f5bcR1101

Just an idea to cut down on the comment / mental overhead here, feel free to ignore.

Reply:
The idea being that the caller is already just glue code to use a `[0, s]` function to generate something in `[l, h]`, so why not go the whole way and change the glue function to use a `[0, s)` generation function instead?

Reply:
In your suggestion, you call with `t - b + 1` but using a finite `Word` we can have

```
Prelude Data.Word> (maxBound::Word32) - (minBound::Word32) + 1
0
```

Are you suggesting we do the guard at the calling site also?

Reply:
> Are you suggesting we do the guard at the calling site also?

Yes, I think that would be consistent. That means `unbiasedWordMult32` has (potentially unchecked) preconditions, which to my mind is fine given that it is a private function with a single caller. I think that the advantage of having the implementation of the algorithm itself be closer to the literature is worth it. But that's really just my opinion / aesthetic view.

Reply:
Done